src/Device/Driver/LX_EOS/LXEosDownload.cpp

```diff
@@ -0,0 +1,245 @@
+  if (flight_count > flight_list.size())
+    flight_count = flight_list.size();
+
+  env.SetProgressRange(flight_count + 1);
+  env.SetProgressPosition(1);
+
+  if (success) {
+    for (uint8_t i = 0; i < flight_count; i++) {
+      RecordedFlightInfo flight;
+
+      // Try up to 5 times before giving up
+      for (uint8_t attempt = 0; attempt < 5; attempt++) {
+        success = GetFlightInfo(i + 1, flight, env);
+        if (success)
+          break;
+      }
+
+      if (!success)
+        break;
+
+      flight_list.append(flight);
+      env.SetProgressPosition(i + 2);
+    }
+  }
+
+  env.SetProgressPosition(flight_count + 1);
+
+  port.StartRxThread();
+  return success;
+}
+
+bool
+LXEosDevice::DownloadFlight(const RecordedFlightInfo& flight,
+                            Path path,
+                            OperationEnvironment& env)
+{
+  FileOutputStream fos(path);
+  BufferedOutputStream bos(fos);
+
+  bool success = true;
+
+  port.StopRxThread();
+
+  uint16_t flight_id = flight.internal.lx_eos.flight_id;
+  uint32_t bytes_remaining = flight.internal.lx_eos.file_size;
+
+  env.SetProgressRange(100);
+  env.SetProgressPosition(0);
+
+  // Download blocks until remaining bytes reach zero
+  for (uint16_t block_id = 0; bytes_remaining > 0; block_id++) {
+    AllocatedArray<std::byte> block;
+
+    // Try up to 5 times before giving up
+    for (uint8_t attempt = 0; attempt < 5; attempt++) {
+      try {
+        block = GetFlightLogBlock(flight_id, block_id, env);
+      } catch (...) {
+        success = false;
+        continue;
+      }
+      success = true;
+      break;
+    }
+
+    if (!success)
+      break;
+
+    bos.Write(block);
+
+    // Make sure that bytes remaining cannot underflow
+    if (block.size() > bytes_remaining)
+      return false;
+
+    bytes_remaining -= block.size();
+
+    // Progress in percents downloaded
+    float progress =
+      100.0f * (1.0f - static_cast<float>(bytes_remaining) /
+                         static_cast<float>(flight.internal.lx_eos.file_size));
+    env.SetProgressPosition(static_cast<unsigned int>(floor(progress)));
+  }
+
+  port.Flush();
+
+  if (success) {
+    bos.Flush();
+    fos.Commit();
+  }
+
+  env.SetProgressPosition(100);
+
+  port.StartRxThread();
+  return success;
+}
+
+bool
+LXEosDevice::GetNumberOfFlights(uint8_t& flight_count,
+                                OperationEnvironment& env)
+{
+  // Request
+  EosGetNumOfFlights data;
+  auto data_array = reinterpret_cast<std::byte*>(&data);
+  std::span<std::byte> message(data_array, sizeof(data));
+  if (!WriteAndWaitForACK(message, env))
+    return false;
+
+  // Read response (ACK byte already read)
+  EosNumOfFlightsResponse response;
+  try {
+    port.FullRead(
+      { reinterpret_cast<std::byte*>(&response) + 1, sizeof(response) - 1 },
+      env,
+      std::chrono::milliseconds(5000));
+  } catch (...) {
+    return false;
```

Comment:
Again: what's the point of this exception handler All it does is discard details about the error condition, but why?